.github/workflows/integrate.yaml

```diff
@@ -0,0 +1,75 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install tox
+    - name: Lint code
+      run: tox -vve lint
+
+  unit:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install dependencies
+        run: pip install tox
+      - name: Run unit tests
+        run: tox -vve unit
+
+  integration:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Setup operator environment
+      uses: charmed-kubernetes/actions-operator@main
+      with:
+        provider: microk8s
+
+    # TODO: Remove once the actions-operator does this automatically
+    - name: Configure kubectl
+      run: |
+        sg microk8s -c "microk8s config > ~/.kube/config"
+
+    - name: Test
+      run: |
+        sg microk8s -c "tox -vve integration -- --model testing"
+
+      # On failure, capture debugging resources
```

Comment:
This one I actually like having.  fwiw, I actually misread this the first time I saw it in charms.  I see that comment as a header for the next 6 things, all of which happen in failures.  Maybe it could be formatted differently instead?